windows/wsl_distro_install_uninstall/run_wsl_distro.yml

```diff
@@ -0,0 +1,60 @@
+# Copyright 2023 VMware, Inc.
+# SPDX-License-Identifier: BSD-2-Clause
+---
+# Description:
+# Run WSL distribution
+
+# List the installed distribution
+- name: "List installed WSL distribution"
+  ansible.windows.win_shell: "wsl --list | Out-File -FilePath {{ wsl_file_path_win }}\\wsl_list_distro.txt"
+  register: win_powershell_cmd_output
+  ignore_errors: true
+  delegate_to: "{{ vm_guest_ip }}"
+  ignore_unreachable: true
+  until:
+    - win_powershell_cmd_output.rc is defined
+    - win_powershell_cmd_output.rc == 0
+  retries: 5
+  delay: 60
+
+- name: "Process the WSL output message"
+  include_tasks: wsl_convert_binary_to_txt.yml
+  vars:
+    source_path: "{{ wsl_file_path_win }}\\wsl_list_distro.txt"
+    file_name: "wsl_list_distro.txt"
+
+- name: "Validate the WSL list command execution result"
+  ansible.builtin.assert:
+    that:
+      - lookup('ansible.builtin.file', wsl_new_file) is search(wsl_distribution_name)
+    fail_msg: "The WSL distribution {{ wsl_distribution_name }} is not installed."
+
+# Execute command on the distribution
+- name: "Set the command executed on WSL distribution "
+  ansible.builtin.set_fact:
+    distro_command: "cat /etc/os-release"
+
+- name: "Process the WSL output message"
+  include_tasks: ../utils/win_execute_cmd.yml
+  vars:
+    win_powershell_cmd: "wsl -d {{ wsl_distribution_name }} {{ distro_command}} | Out-File {{ wsl_file_path_win }}\\wsl_execute_cmd.txt"
+    win_execute_cmd_ignore_error: true
+
+- name: "Process the WSL output message"
+  include_tasks: wsl_convert_binary_to_txt.yml
+  vars:
+    source_path: "{{ wsl_file_path_win }}\\wsl_execute_cmd.txt"
+    file_name: "wsl_execute_cmd.txt"
+
+- name: "Check the execution result of WSL distribution command"
+  ansible.builtin.assert:
+    that:
+      - win_powershell_cmd_output.rc is defined
+      - win_powershell_cmd_output.rc == 0
```

Comment:
I have a question about checking powershell rc after converting binary to txt. What if the powershell failed and there is not file created from powershell command? I think it's better to check rc before converting the output file.

Reply:
if check the rc firstly. Then there's no chance to convert the output message. The wsl output is special which is not human readable directly. 
E.g
"stderr": "A\u0000c\u0000c\u0000e\u0000s\u0000s\u0000 \u0000i\u0000s\u0000 \u0000d\u0000e\u0000n\u0000i\u0000e\u0000d\u0000.\u0000\r\u0000\r\u0000\n\u0000", "stderr_lines": ["A\u0000c\u0000c\u0000e\u0000s\u0000s\u0000 \u0000i\u0000s\u0000 \u0000d\u0000e\u0000n\u0000i\u0000e\u0000d\u0000.\u0000", "\u0000", "\u0000", "\u0000"], "stdout": "I\u0000n\u0000s\u0000t\u0000a\u0000l\u0000l\u0000i\u0000n\u0000g\u0000:\u0000 \u0000V\u0000i\u0000r\u0000t\u0000u\u0000a\u0000l\u0000 \u0000M\u0000a\u0000c\u0000h\u0000i\u0000n\u0000e\u0000 \u0000P\u0000l\u0000a\u0000t\u0000f\u0000o\u0000r\u0000m\u0000\r\u0000\r\u0000\n\u0000V\u0000i\u0000r\u0000t\u0000u\u0000a\u0000l\u0000 \u0000M\u0000a\u0000c\u0000h\u0000i\u0000n\u0000e\u0000 \u0000P\u0000l\u0000a\u0000t\u0000f\u0000o\u0000r\u0000m\u0000 \u0000h\u0000a\u0000s\u0000 \u0000b\u0000e\u0000e\u0000n\u0000 \u0000i\u0000n\u0000s\u0000t\u0000a\u0000l\u0000l\u0000e\u0000d\u0000.\u0000\r\u0000\r\u0000\n\u0000D\u0000o\u0000w\u0000n\u0000l\u0000o\u0000a\u0000d\u0000i\u0000n\u0000g\u0000:\u0000 \u0000W\u0000i\u0000n\u0000d\u0000o\u0000w\u0000s\u0000 \u0000S\u0000u\u0000b\u0000s\u0000y\u0000s\u0000t\u0000e\u0000m\u0000 \u0000f\u0000o\u0000r\u0000 \u0000L\u0000i\u0000n\u0000u\u0000x\u0000\r\u0000\r\u0000\n\u0000I\u0000n\u0000s\u0000t\u0000a\u0000l\u0000l\u0000i\u0000n\u0000g\u0000:\u0000 \u0000W\u0000i\u0000n\u0000d\u0000o\u0000w\u0000s\u0000 \u0000S\u0000u\u0000b\u0000s\u0000y\u0000s\u0000t\u0000e\u0000m\u0000 \u0000f\u0000o\u0000r\u0000 \u0000L\u0000i\u0000n\u0000u\u0000x\u0000\r\u0000\r\u0000\n\u0000",